src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
+
+### JSON Web Token signatures
+
+Blockstack's authentication tokens are based on the [ RFC 7519 OAuth JSON Web Token (JWT)](https://tools.ietf.org/html/rfc7519) with additional support for the secp256k1 curve used by bitcoin and many other cryptocurrencies.
+
+This support is indicated by specifying `ES256K` in the `alg` key to indicated that the JWT signature uses ECDSA using the secp256k1 curve.  We provide both [JavaScript](https://github.com/blockstack/jsontokens-js) and [Ruby](https://github.com/blockstack/ruby-jwt-blockstack/tree/ruby-jwt-blockstack) JWT libraries with support for this curve.
+
+### Authentication request payload schema
+
+``` JavaScript
+const requestPayload = {
+    jti, // UUID
+    iat, // JWT creation time in seconds
+    exp, // JWT expiration time in seconds
+    iss, // legacy decentralized identifier generated from transit key
+    public_keys, // single entry array with public key of transit key
+    domain_name, // app origin
+    manifest_uri, // url to manifest file - must be hosted on app origin
+    redirect_uri, // url to which browser redirects user on auth approval - must be hosted on app origin
+    version, // version tuple
+    do_not_include_profile, // a boolean flag asking browser to send profile url instead of profile object
+    supports_hub_url, // a boolean flag indicating gaia hub support
+    scopes // an array of string values indicating scopes requested by the app
+  }
 ```
+ 
+  
+### Authentication response payload schema
+  
+```JavaScript
+    const responsePayload = {
+    jti, // UUID
+    iat, // JWT creation time in seconds
+    exp, // JWT expiration time in seconds
+    iss, // legacy decentralized identifier (string prefix + identity address) - this uniquely identifies the user
+    private_key, // encrypted private key payload
+    public_keys, // single entry array with public key
+    profile, // profile object or null if passed by profile_url
+    username, // blockstack id username (if any)
+    core_token, // encrypted core token payload
+    email, // email if email scope is requested & email available
+	  profile_url, // url to signed profile token
+    hubUrl, // url pointing to user's gaia hub
+    version // version tuple
+  }
+```
+  
+## `blockstack:` custom protocol handler
+
+The `blockstack:` custom protocol handler is how Blockstack apps send their authentication requests to the Blockstack Browser. When the Blockstack Browser is installed on a user's computer, it registers itself as the handler for the `blockstack:` customer protocol.
```

Comment:
Blockstack apps send their authentication requests to the Blockstack Browser via the `blockstack:` custom protocol handler. This protocol handler is registered when Blockstack Browser is installed.